core/compiler.cpp

```diff
@@ -180,8 +189,32 @@ partitioning::GraphAndMapping BuildHybridGraph(
             true);
 
         seg_block.update_graph(temp_g);
+      } else {
+        num_torch_segments++;
+
+        // If full compilation is expected, ensure that all operators in Torch blocks are
+        // for collections processing
+        if (expect_full_compilation) {
+          for (auto torch_node : seg_block.block()->nodes()) {
+            if (partitioning::CollectionSchemas.find(torch_node->kind().toQualString()) ==
+                partitioning::CollectionSchemas.end()) {
+              LOG_WARNING(
+                  "Full compilation specified but node " << torch_node->kind().toQualString()
+                                                         << " was executed in Torch.");
```

Comment:
Could potentially be a `LOG_ERROR` to increase severity or `TORCHTRT_CHECK` to completely halt compilation, since entering this `if` block indicates that the user specified `require_full_compilation=True`, but an operation was set to run in Torch which is not for the purpose of creating the collection.

Reply:
I dont think we need this warning, it could probably be debug if all that is happening is collection packing/unpacking. But I think our criteria is more restrictive. It there should not really be any intermediate packing unpacking imo. Maybe there are some counter examples?

Reply:
Since this message should only display if a non-collection-based operation was determined to run in Torch (which should not occur - this violates the `require_full_compilation=True` condition), I have made this log command a `LOG_ERROR`, similarly to the case below.

Reply:
If its an error then it should use THROW_ERROR and quit instead of continuing